core/src/main/scala/org/apache/pekko/persistence/cassandra/reconciler/AllTags.scala

```diff
@@ -36,16 +36,13 @@ private[pekko] final class AllTags(session: ReconciliationSession) {
     session
       .selectAllTagProgress()
       .map(_.getString("tag"))
-      .statefulMapConcat(() => {
-        var seen = Set.empty[String]
-        tag =>
-          if (!seen.contains(tag)) {
-            seen += tag
-            List(tag)
-          } else {
-            Nil
-          }
-      })
+      .statefulMap(() => Set.empty[String])(
+        (seen, tag) => {
+          val element = if (seen.contains(tag)) None else Some(tag)
+          (seen + tag, element)
```

Comment:
`seen + tag` may allocate a new object when the tag has been seen, can we optimize this?

Reply:
is using mutable.Set ok? - that avoids the creation of extra Set instances